File: apps/dotnet/Functions/StorageProbeFunction.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Storage connectivity check — verifies storage account configuration.
/// Route: GET /api/storage/probe
/// </summary>
public class StorageProbeFunction
{
    private readonly ILogger<StorageProbeFunction> _logger;

    public StorageProbeFunction(ILogger<StorageProbeFunction> logger)
    {
        _logger = logger;
    }

    [Function("storageProbe")]
    public IActionResult Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "storage/probe")] HttpRequest req)
    {
        var storageConn = System.Environment.GetEnvironmentVariable("AzureWebJobsStorage");
        var hasStorage = !string.IsNullOrEmpty(storageConn);

        _logger.LogInformation("Storage probe: connection {Status}",
            hasStorage ? "configured" : "missing");

        string connectionType;
        if (!hasStorage)
            connectionType = "none";
        else if (storageConn!.Contains("AccountName="))
            connectionType = "connectionString";
        else
            connectionType = "identity";

        return new OkObjectResult(new
        {
            storageConfigured = hasStorage,
            connectionType
        });
    }
}

File: apps/java/src/main/java/com/functions/StorageProbeFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import java.util.Optional;

/**
 * Storage probe — checks storage account connectivity.
 */
public class StorageProbeFunction {

    @FunctionName("storageProbe")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "storage/probe")
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {

        String storageConn = System.getenv("AzureWebJobsStorage");
        boolean hasStorage = storageConn != null && !storageConn.isEmpty();

        context.getLogger().info("Storage probe: connection "
            + (hasStorage ? "configured" : "missing"));

        String body = String.format(
            "{\"storageConfigured\":%s,\"connectionType\":\"%s\"}",
            hasStorage,
            hasStorage ? (storageConn.contains("AccountName=") ? "connectionString" : "identity") : "none");

        return request.createResponseBuilder(HttpStatus.OK)
                .header("Content-Type", "application/json")
                .body(body)
                .build();
    }
}
